### PR TITLE
1097 bidversion

### LINKED ISSUE #1097: As a sequencer i should be able to reject bids that are of an incompatible version

### PR DESCRIPTION
**Pull request by sanchaymittal**
## Type of change

- [ ] Docs change / dependency upgrade
- [X] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes #1097

### CODE REVIEW

File: packages/agents/sequencer/src/config.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ import { SequencerConfig, SequencerConfigSchema } from "./lib/entities";
 const MIN_SUBGRAPH_SYNC_BUFFER = 25;
 const DEFAULT_AUCTION_WAIT_TIME = 30_000;
 
+const DEFAULT_SUPPORTED_BID_VERSION = "0.2.0-beta.0";
```
Comment (rhlsthrm) [inline packages/agents/sequencer/src/config.ts:11]: Should default be its own package.json?

File: packages/agents/sequencer/src/lib/operations/auctions.ts [hunk 2]
```diff
@@ -39,6 +41,15 @@ export const storeBid = async (bid: Bid, _requestContext: RequestContext): Promi
     });
   }
 
+  // check if bid router version is compatible with hosted sequencer
+  const checkVersion = compareVersions(bid.routerVersion, config.supportedBidVersion!);
```
Comment (rhlsthrm) [inline packages/agents/sequencer/src/lib/operations/auctions.ts:45]: So we will error on any version mismatch? Do we want to use `~` to support patch versions? https://www.npmjs.com/package/compare-versions

**Review by rhlsthrm — APPROVED** [review]

File: packages/agents/sequencer/src/config.ts (hunk 1 above)
Comment (sanchaymittal) [reply packages/agents/sequencer/src/config.ts:11]: hmm, okay!

File: packages/agents/sequencer/src/lib/operations/auctions.ts (hunk 2 above)
Comment (sanchaymittal) [reply packages/agents/sequencer/src/lib/operations/auctions.ts]: updated in human-readable form

**Review by rhlsthrm — APPROVED** [review]

**Pull request merged**